app/WebParsing/UtsgJsonParser.hs

```diff
@@ -1,83 +1,95 @@
-                                insertMany_, runSqlite, selectFirst, (==.), (=.))
-import Database.Tables (Courses (..), EntityField (..), MeetTime (..), Meeting (..),
-                        Times (..), buildTimes)
-import Network.HTTP.Conduit (simpleHttp)
-
-coursesJson :: FilePath
-coursesJson = "courses.json"
-
-parseAllCourses :: IO ()
-parseAllCourses = do
-    runSqlite databasePath parseCourses
-
-parseCourses :: SqlPersistM ()
-parseCourses = do
-    deleteWhere ([] :: [Filter Times])
-    deleteWhere ([] :: [Filter Meeting])
-    liftIO . print $ T.pack "parsing JSON data"
-    resp <- liftIO $ decodeFileStrict coursesJson
-    let coursesLst :: Maybe (HM.HashMap T.Text (Maybe DB)) = resp
-        courseData = maybe [] (map dbData . catMaybes . HM.elems) coursesLst
-        (_, sections) = unzip courseData
-        meetings = concat sections
-    mapM_ insertMeeting meetings
+import Database.Persist.Sqlite (SqlPersistM, Update, Entity, entityKey, entityVal, deleteWhere, upsert,
+                                insert, insertMany_, runSqlite, selectFirst, selectList, (==.), (=.))
+import Database.Tables (Courses (..), EntityField (..), MeetTime (..), Meeting (..), buildTimes)
+import Network.HTTP.Conduit
 
 -- | Parse all timetable data.
 parseTimetable :: IO ()
 parseTimetable = do
     orgs <- getOrgs
     runSqlite databasePath $ mapM_ insertAllMeetings orgs
 
--- | Return a list of all the "orgs" in FAS. These are the values which can be
---   passed to the timetable API with the "org" key.
+-- | Get all the orgs from the Courses schema in the database
 getOrgs :: IO [T.Text]
-getOrgs = do
-    resp <- simpleHttp orgApiUrl
-    let rawJSON :: Maybe (HM.HashMap T.Text Object) = decode resp
-    return $ maybe [] (concatMap $ map toText . KM.keys) rawJSON
+getOrgs = runSqlite databasePath $ do
+    courseEntities <- selectList [] [] :: SqlPersistM [Entity Courses]
+    let courseCodes = map (coursesCode . entityVal) courseEntities
+    let orgsSet = Set.fromList $ map (T.take 3) courseCodes
+    return $ Set.toList orgsSet
 
--- | Retrieve and store all timetable data for the given department.
+-- | insert/update all the data into the Meeting and Times schema by creating and sending
+--   the http request to Artsci Timetable and then parsing the JSON response
 insertAllMeetings :: T.Text -> SqlPersistM ()
 insertAllMeetings org = do
     liftIO . print $ T.append "parsing JSON data from: " org
-    resp <- liftIO . simpleHttp $ T.unpack (T.append timetableApiUrl org)
-    let coursesLst :: Maybe (HM.HashMap T.Text (Maybe DB)) = decode resp
-        courseData = maybe [] (map dbData . catMaybes . HM.elems) coursesLst
-        -- courseData contains courses and sections;
-        -- only sections are currently stored here.
-        (_, sections) = unzip courseData
-        meetings = concat sections
-    mapM_ insertMeeting meetings
+
+    -- Define the request
+    let reqBody = object [ "campuses" .= ([] :: [T.Text])
```

Comment:
This value should be defined as a separate value in the config file, since it's really tied to the API, similar to the url being used. (You can make the value a function which takes in an `org`.) Similarly for the request headers.

Also, you need to make changes to both `Config.hs` and `DevelopmentConfig.hs`; only the latter is committed to the repo.

Reply:
For making this change I had to add a dependency as there was a type mismatch which could only be resolved with adding the HTTP TYPES.